matrix/buffer.py

```diff
@@ -640,22 +652,43 @@ def _get_prefix_color(prefix):
 
     def _add_user_to_nicklist(self, user):
         # type: (WeechatUser) -> None
-        nick_pointer = W.nicklist_search_nick(self._ptr, "", user.nick)
+        if user.nick in self.duplicate_nicks:
```

Comment:
The disambiguation algorithm works well, great work. Sadly we forgot a crucial feature about it, the room display name depends on user display names if the room is unnamed.

Therefore the algorithm should go into nio, and weechat should ask nio when adding, removing, and changing nicks if there are conflicts. After a user is added, removed or a nick change is performed weechat needs to ask nio for the new room display name and set the buffers short_name property to that (it already does that to an extent).

Reply:
do you want nio to handle disambiguation in full or just put `get_disambiguated_nick` into nio? I kinda have a feeling that nio is a bit more low-level to handle that...
Also I still feel it as a bit of controversy, I know that it is recommended by spec but I really want to avoid using fully disambiguated nicks in weechat room buffer (not in nicklist) since it makes output unwieldy or nick get cut by `weechat.look.prefix_align_max`.

Reply:
I'm not sure what you mean by full. Anything that is needed to calculate a disambiguated display name and the room name should go into nio. The client of course can pick and chose what it will display to the user. But the client should be able to ask nio for a disambiguated name. 

So weechat remembers what it has displayed (it already does that), upon a join/leave or display name change it checks if it would have a conflict and if so asks nio for the disambiguated names.

The room name calculation is inside of nio, I don't see how room member name calculation is any higher or lower level than this. Especially since correct room name calculation depends on the room member names.

If you want to mix and match what gets displayed please make it configurable, I have a feeling that people will have different opinions on that. So 3 modes I guess, what we currently have (user ids without the host part), full display names, and your mixed mode.

Reply:
Sorry if it wasn't so clear, by full I mean that nio should track collisions and return disambiguated name only in case when there is a clash in a single the room. I understand that for room names this is kinda required.

Reply:
Frankly speaking if poljar/matrix-nio#4 going to be merged current mode of using part of user_id without host for nicks would be inconsistent with automatic room names. Do you have a suggestion how to make automatic room names configurable in matrix-nio?

Reply:
Maybe I'm missing something... Could you elaborate on why it would be inconsistent?

Reply:
In case if you have a chat with account with display name "Foo" and user id "@bar:example.org" and if poljar/matrix-nio#4 would be merged `named_room_name` is going to return room name as "Foo" but weechat nicklist going to contain "bar" in nicklist instead of "Foo" which can be confusing.

Reply:
The room names are already inconsistent since they use the full user id and the nicklist the short/local part of the user id.

That being said a function that uses user ids instead of display names for room name calculation can be added to nio and the old behavior can be preserved that way. Another function that gives a room name consisting of only the local part of the user id is fine as well.

Reply:
@gordon-quad me? what?

Reply:
sorry, github falsly triggered on part of matrix user id `@bar:example.org`